### PR TITLE
fix lazy value in ir_analyze_instruction_elem_ptr

### LINKED ISSUE #4954: Assertion in `bigint_as_unsigned`: caused by `&buf[@sizeOf(...)]`
Minimal repro:

```zig
fn f(buf: []u8) void {
    var ptr = &buf[@sizeOf(u32)]; // causes the compiler to die
}

pub fn main() void {
    var buf: [4096]u8 = undefined;
    f(&buf);
}
```
```
Unreachable at [REDACTED]\src\bigint.cpp:1677 in bigint_as_unsigned. This is a bug in the Zig compiler.
Unable to dump stack trace: debug info stripped
```

### PR DESCRIPTION
**Pull request by xackus**
Fixes #4954.

### CODE REVIEW

File: src/ir.cpp [hunk 1]
```diff
@@ -21049,6 +21049,12 @@ static IrInstGen *ir_analyze_instruction_elem_ptr(IrAnalyze *ira, IrInstSrcElemP
 
     bool safety_check_on = elem_ptr_instruction->safety_check_on;
     if (instr_is_comptime(casted_elem_index)) {
+        if ((err = ir_resolve_const_val(ira->codegen, ira->new_irb.exec,
```
Comment (LemonBoy) [inline src/ir.cpp:21052]: Small nit, the pattern we usually use is:
```zig
        ZigValue *val = ir_resolve_const(ira, casted_elem_index, UndefBad);
        if (val == nullptr)
            return ira->codegen->invalid_inst_gen;
```

Comment (xackus) [reply src/ir.cpp:21052]: thanks

**Comment by andrewrk** [comment]
The fix looks good. Since a minimal repro was provided, can you turn the minimal repro into a test case?

**Comment by xackus** [comment]
Already did that, just forgot to commit and push.

**Pull request merged**